docs/content/en/docs/about/under-the-hood/guarantees.md

```diff
@@ -15,12 +15,20 @@ see which components support specific guarantees.
 
 ### At-least-once
 
-The **at-least-once** delivery guarantee ensures that an [event] received by a Vector component is
-ultimately delivered at least once. While rare, it is possible for an event to be delivered more
-than once. See the [Does Vector support exactly-once delivery?](#faq-at-least-once) FAQ below).
+The **at-least-once** delivery guarantee ensures that an [event]
+received by a Vector component is ultimately delivered at least
+once. For a source, this indicates that it will wait for _all_ connected
+sinks to mark the event as delivered before acknowledging receipt of the
+event. For a sink, this indicates that it will attempt to retry the
+delivery until the events are either accepted or rejected and then
+signal the source with the results of that delivery.
+
+While rare, it is possible for an event to be delivered more than
+once. See the [Does Vector support exactly-once
+delivery?](#faq-at-least-once) FAQ below).
```

Comment:
Should we add that the guarantee only applies if end-to-end acknowledgements are enabled?

Reply:
Ah yes, agreed.